pkg/storaged/client.js

```diff
@@ -372,44 +373,65 @@
                 });
         }
 
-        function enable_features() {
-            return enable_udisks_features().then(function (features) {
-                return cockpit.spawn([ "which", "vdo" ], { err: "ignore" }).then(
-                    function () {
-                        features.vdo = true;
-                        client.vdo_overlay.start();
-                        return features;
-                    },
-                    function () {
-                        return features;
-                    });
-            });
-        }
-
         function enable_udisks_features() {
             if (!client.manager.valid)
-                return cockpit.resolve(false);
+                return cockpit.resolve();
             if (!client.manager.EnableModules)
-                return cockpit.resolve({ });
+                return cockpit.resolve();
             return client.manager.EnableModules(true).then(
                 function() {
                     var defer = cockpit.defer();
                     client.manager_lvm2 = proxy("Manager.LVM2", "Manager");
                     client.manager_iscsi = proxy("Manager.ISCSI.Initiator", "Manager");
                     wait_all([ client.manager_lvm2, client.manager_iscsi],
                             function () {
-                                var iscsi = (hacks.with_storaged_iscsi_sessions != "no" &&
-                                        client.manager_iscsi.valid &&
-                                        client.manager_iscsi.SessionsSupported !== false);
-                                defer.resolve({ lvm2: client.manager_lvm2.valid, iscsi: iscsi });
+                                client.features.lvm2 = client.manager_lvm2.valid;
+                                client.features.iscsi = (hacks.with_storaged_iscsi_sessions != "no" &&
+                                                         client.manager_iscsi.valid &&
+                                                         client.manager_iscsi.SessionsSupported !== false);
+                                defer.resolve();
                             });
                     return defer.promise;
                 }, function(error) {
                     console.warn("Can't enable storaged modules", error.toString());
-                    return cockpit.resolve({ });
+                    return cockpit.resolve();
                 });
         }
 
+        function enable_vdo_features() {
+            return cockpit.spawn([ "which", "vdo" ], { err: "ignore" }).then(
+                function () {
+                    client.features.vdo = true;
+                    client.vdo_overlay.start();
+                    return cockpit.resolve();
+                },
+                function () {
+                    return cockpit.resolve();
+                });
```

Comment:
No need to wrap a function around this, you can just directly give `cockpit.resolve` as second argument (same below).

Reply:
But then `this` will be different by the time `cockpit.resolve` is actually called, no?  It doesn't matter in this case, but that difference and the fact that it breaks symmetry between the two cases makes me prefer the long form actually.